docs/Utilities/Container.md

```diff
@@ -1,8 +1,20 @@
 # Container
 
-Each extension point is provided a container API which provides additional methods the extension can use to communicate with Shopify Admin.
+Containers are interface elements that are used to render your extension with Argo components. Your extension can use the **container API** to configure its behaviour and appearance inside of Shopify Admin.
 
-## Subscription Management
+## Product Subscription
+
+There are two different container types for the Product subscription extension, depending on the extension mode:
+
+#### Add, Remove
+**App modal container**: A smaller overlay typically used for editing an existing configuration or confirming a change. In the app modal container, primary and secondary action buttons are responsible for triggering a custom behaviour, such as performing a resource update.
+
+#### Create, Edit
+**App overlay container**: A full-screen container designed for complex tasks, like configuring a subscription for the first time.
+
+### Container API
+
+#### Add, Remove (Modal container)
```

Comment:
I think this info should be found in Product Subscription page and all the relevant information about it is collocated as to not bloat the Container.md and Data.md with specific implemtnation in the future.